File: language-analysis/2.1.Concepts-Implementation/G-Concurrency/multiple_async_tasks.py
import asyncio


async def task(name, delay):
    await asyncio.sleep(delay)
    print(f"Task {name} finished")


async def run_async_examples():
    await asyncio.gather(
        task("A", 1),
        task("B", 2),
    )


asyncio.run(run_async_examples())

File: language-analysis/2.1.Concepts-Implementation/G-Concurrency/multiple_async_tasks.js
function task(name, delay) {
  return new Promise((resolve) => {
    setTimeout(() => {
      console.log(`Task ${name} finished`);
      resolve();
    }, delay);
  });
}

(async () => {
  await Promise.all([
    task("A", 500),
    task("B", 1000),
  ]);
})();
